crates/sui-json-rpc-api/src/rpc_types.rs

```diff
@@ -1190,6 +1190,18 @@ pub struct OwnedObjectRef {
     pub reference: SuiObjectRef,
 }
 
+#[serde_as]
+#[derive(Eq, PartialEq, Clone, Debug, Serialize, Deserialize, JsonSchema)]
+#[serde(rename = "EventEnvelope", rename_all = "camelCase")]
+pub struct SuiEventEnvelope {
+    /// UTC timestamp in milliseconds since epoch (1/1/1970)
+    pub timestamp: u64,
+    /// Transaction digest of associated transaction, if any
+    pub tx_digest: Option<TransactionDigest>,
```

Comment:
I think we should, and I think we can use a const TransactionDigest for system events, so we don't have to deal with Options.

Reply:
I'm not sure about that.  We should check with others including @lxfind and @gdanezis but here's my take on it:

I don't think we should manufacture TX digests for events which aren't tied to transactions, because TX digests have a specific meaning: they refer to valid transactions which are certified.  If there could be TX digests which are "fake" then we need to explain this, and have special cases in our code to deal with it.   Customers must also deal with this.  If a customer sees a TX digest for a "checkpoint event", how will they interpret it?  Will they think, gee this must point to the last TX before the checkpoint?   They might then try to use our APIs to retrieve such a transaction, only to find there is no such transaction.   This seems like a bad idea.

Maybe, if that digest was all zeroes, then people might think gee this is something different, but I'm not sure that makes it better.